python/nav/statemon/checker/HttpChecker.py

```diff
@@ -77,8 +78,9 @@ def execute(self):
             i.putheader('User-Agent',
                         'NAV/servicemon; version %s' % buildconf.VERSION)
             if username:
-                auth = "%s:%s" % (username, password)
-                i.putheader("Authorization", "Basic %s" % auth.encode("base64"))
+                auth = "{}:{}".format(username, password).encode("utf-8")
+                auth = base64.b64encode(auth).decode("utf-8")
```

Comment:
Clunky, but  I don't know any better solution.

Reply:
Yes, I really don't understand why the result of a `b64encode` needs to a bytes object, it should always be printable ascii anyway...

Reply:
`b64encode()` encodes bytes to ASCII text so it makes sense that I takes a `bytes` object. What puzzles me is why it also returns a `bytes` object. Clearly it should return a `str`.